app/assets/javascripts/admin/products/services/variant_unit_manager.js.coffee

```diff
@@ -60,8 +72,13 @@ angular.module("admin.products").factory "VariantUnitManager", (availableUnits)
 
     @compatibleUnitScales: (scale, unitType) ->
       scaleSystem = @units[unitType][scale]['system']
-      (parseFloat(scale) for scale, scaleInfo of @units[unitType] when scaleInfo['system'] == scaleSystem).sort (a, b) ->
-         a - b
+      if availableUnits
+        available = availableUnits.split(",")
+        (parseFloat(scale) for scale, scaleInfo of @units[unitType] when scaleInfo['system'] == scaleSystem and available.includes(scaleInfo['name'])).sort (a, b) ->
+          a - b
+      else
+        (parseFloat(scale) for scale, scaleInfo of @units[unitType] when scaleInfo['system'] == scaleSystem).sort (a, b) ->
+          a - b
```

Comment:
It looks like the condition could have been included into the existing code, rather than copy to a separate branch. 
In fact, it looks like all functions in this class need to filter by available units.

I think there should be a filtered `units` list, which would probably be shared with the other functions.
But I have to admit I'm struggling with this kind of Javascript and am not sure how to refactor it!

So if you'd like to try, I'd be very thankful, but don't expect you to!